### PR TITLE
PPS tests updated with 2024 data

### PR DESCRIPTION
**Pull request by grzanka**
#### PR description:

This PR updates tests (relvals and unittest) running reconstruction for PPS detectors. Main reason for update is to have the tests running on current 2024 data. See also relevant issue: https://github.com/cms-sw/cmssw/issues/44095

#### PR validation:

`scram b runtests` for `RecoPPS\Local` and standard `runTheMatrix` with newly introduced relvals 1046, 1047, 1048 and 1049.

### CODE REVIEW

**Comment by perrotta** [comment]
test parameters:
- workflows = 1046,1047,1048

**Comment by perrotta** [comment]
please test

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -471,6 +471,9 @@
 #### PPS run3 2022B
 steps['RunRawPPS2022B']={'INPUT':InputInfo(dataSet='/AlCaPPS/Run2022B-v1/RAW',label='2022B',events=100000,location='STD', ls={355207: [[1, 100]]})}
 steps['RunRawPPS2022A']={'INPUT':InputInfo(dataSet='/AlCaPPS/Run2022A-v1/RAW',label='2022A',events=100000,location='STD', ls={354332: [[1, 100]]})}
+#### PPS run3 2024E
+steps['RunRawPPS2024E']={'INPUT':InputInfo(dataSet='/AlCaPPSPrompt/Run2024D-v1/RAW',label='2024E',events=100000,location='STD', ls={380470: [[1, 100]]})}
+steps['RunRawPPSReco2024E']={'INPUT':InputInfo(dataSet='/ZeroBias/Run2024E-v1/RAW',label='2024E',events=100000,location='STD', ls={381380: [[1, 100]]})}
```
Comment (perrotta) [inline Configuration/PyReleaseValidation/python/relval_steps.py]:
```suggestion
#### PPS run3 2024D
steps['RunRawPPS2024D']={'INPUT':InputInfo(dataSet='/AlCaPPSPrompt/Run2024D-v1/RAW',label='2024D',events=100000,location='STD', ls={380470: [[1, 100]]})}
#### PPS run3 2024E
steps['RunRawPPSReco2024E']={'INPUT':InputInfo(dataSet='/ZeroBias/Run2024E-v1/RAW',label='2024E',events=100000,location='STD', ls={381380: [[1, 100]]})}
```

File: Configuration/PyReleaseValidation/python/relval_production.py [hunk 2]
```diff
@@ -31,9 +31,11 @@
 
 workflows[1041] = ['',['RunExpressPhy2017F','TIER0EXPPPSCAL','ALCASPLITPPSCAL','ALCAHARVDPPSCAL']]
 workflows[1042] = ['',['RunExpressPhy2017F','TIER0EXPPPSCALALIG','ALCASPLITPPSALIG','ALCAHARVDPPSALIG']]
-workflows[1043] = ['',['RunRawPPS2022B','TIER0EXPPPSCALRUN3','ALCASPLITPPSCALSAMPIC','ALCAHARVDPPSCALSAMPIC']]
-workflows[1044] = ['',['RunRawPPS2022B','TIER0EXPPPSCALRUN3','ALCASPLITPPSCALRUN3','ALCAHARVDPPSCALRUN3']]
-workflows[1045] = ['',['RunRawPPS2022B','TIER0EXPPPSCALRUN3','ALCASPLITPPSALIGRUN3','ALCAHARVDPPSALIGRUN3']]
+workflows[1046] = ['',['RunRawPPS2024E','TIER0EXPPPSCALRUN3','ALCASPLITPPSCALSAMPIC','ALCAHARVDPPSCALSAMPIC']]
+workflows[1047] = ['',['RunRawPPS2024E','TIER0EXPPPSCALRUN3','ALCASPLITPPSCALRUN3','ALCAHARVDPPSCALRUN3']]
+workflows[1048] = ['',['RunRawPPS2024E','TIER0EXPPPSCALRUN3','ALCASPLITPPSALIGRUN3','ALCAHARVDPPSALIGRUN3']]
```
Comment (perrotta) [inline Configuration/PyReleaseValidation/python/relval_production.py]:
```suggestion
workflows[1046] = ['',['RunRawPPS2024D','TIER0EXPPPSCALRUN3','ALCASPLITPPSCALSAMPIC','ALCAHARVDPPSCALSAMPIC']]
workflows[1047] = ['',['RunRawPPS2024D','TIER0EXPPPSCALRUN3','ALCASPLITPPSCALRUN3','ALCAHARVDPPSCALRUN3']]
workflows[1048] = ['',['RunRawPPS2024D','TIER0EXPPPSCALRUN3','ALCASPLITPPSALIGRUN3','ALCAHARVDPPSALIGRUN3']]
```

**Review by perrotta — COMMENTED** [review]
Since you are using data from the 2024D era, wouldn't be preferable to flag it accordingly when  you do so?

File: Configuration/PyReleaseValidation/python/relval_production.py (hunk 2 above)
Comment (grzanka) [reply Configuration/PyReleaseValidation/python/relval_production.py]: Already done in https://github.com/cms-sw/cmssw/pull/45456/commits/f8064642d22ea7ec408355106ea4ec246d13e32b

File: Configuration/PyReleaseValidation/python/relval_steps.py (hunk 1 above)
Comment (grzanka) [reply Configuration/PyReleaseValidation/python/relval_steps.py]: Done in https://github.com/cms-sw/cmssw/pull/45456/commits/f8064642d22ea7ec408355106ea4ec246d13e32b

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 3]
```diff
@@ -471,6 +471,9 @@
 #### PPS run3 2022B
 steps['RunRawPPS2022B']={'INPUT':InputInfo(dataSet='/AlCaPPS/Run2022B-v1/RAW',label='2022B',events=100000,location='STD', ls={355207: [[1, 100]]})}
 steps['RunRawPPS2022A']={'INPUT':InputInfo(dataSet='/AlCaPPS/Run2022A-v1/RAW',label='2022A',events=100000,location='STD', ls={354332: [[1, 100]]})}
+#### PPS run3 2024D and 2024E
+steps['RunRawPPS2024D']={'INPUT':InputInfo(dataSet='/AlCaPPSPrompt/Run2024D-v1/RAW',label='2024E',events=100000,location='STD', ls={380470: [[1, 100]]})}
```
Comment (perrotta) [inline Configuration/PyReleaseValidation/python/relval_steps.py:475]:
```suggestion
steps['RunRawPPS2024D']={'INPUT':InputInfo(dataSet='/AlCaPPSPrompt/Run2024D-v1/RAW',label='2024D',events=100000,location='STD', ls={380470: [[1, 100]]})}
```

**Comment by saumyaphor4252** [comment]
please test

**Comment by mandrenguyen** [comment]
+reconstruction

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by srimanob** [comment]
+Upgrade

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**